src/main/java/piece/Elephant.java

```diff
@@ -0,0 +1,83 @@
+package piece;
+
+import java.util.EnumMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import board.Board;
+import board.Position;
+
+public class Elephant extends Piece {
+
+    private static final List<Movement> MOVEMENTS = List.of(
+            Movement.TOP_LEFT_TOP_LEFT_TOP,
+            Movement.TOP_RIGHT_TOP_RIGHT_TOP,
+            Movement.BOTTOM_LEFT_BOTTOM_LEFT_BOTTOM,
+            Movement.BOTTOM_RIGHT_BOTTOM_RIGHT_BOTTOM,
+            Movement.LEFT_LEFT_TOP_LEFT_TOP,
+            Movement.LEFT_LEFT_BOTTOM_LEFT_BOTTOM,
+            Movement.RIGHT_RIGHT_TOP_RIGHT_TOP,
+            Movement.RIGHT_RIGHT_BOTTOM_RIGHT_BOTTOM
+    );
+
+    private static final int MOVEMENT_TOTAL_STEP = 3;
+
+    public Elephant(final Team team) {
+        super(team, PieceType.ELEPHANT);
+    }
+
+    @Override
+    protected Set<Position> getMovablePositions(final Position position, final Board board) {
+        Set<Position> movablePositions = new HashSet<>();
+        Map<Movement, Position> beforeLastStepPositions = moveBeforeLastStep(position, board);
+        if (beforeLastStepPositions.isEmpty()) {
+            throw new IllegalArgumentException("현재 해당 기물이 이동 가능한 곳이 없습니다.");
+        }
+        addMovablePosition(board, movablePositions, beforeLastStepPositions);
+        return movablePositions;
+    }
+
+    @Override
+    public PieceType getType() {
+        return this.pieceType;
+    }
+
+    private Map<Movement, Position> moveBeforeLastStep(final Position position, final Board board) {
+        Map<Movement, Position> moveBeforeLastStepPositions = new EnumMap<>(Movement.class);
+        for (Movement movement : MOVEMENTS) {
+            addBeforeLastStepPosition(position, board, movement, moveBeforeLastStepPositions);
+        }
+        return moveBeforeLastStepPositions;
+    }
+
+    private void addBeforeLastStepPosition(final Position position, final Board board, final Movement movement,
+                                           final Map<Movement, Position> moveBeforeLastStepPositions
+    ) {
+        Position movePosition = position;
+        for (int step = 1; step < MOVEMENT_TOTAL_STEP; step++) {
+            movePosition = movement.applyMovementStep(step, movePosition);
+            if (movePosition.isInValidPosition() || board.isExists(movePosition)) {
+                break;
+            }
+            if (step == (MOVEMENT_TOTAL_STEP - 1)) {
+                moveBeforeLastStepPositions.put(movement, movePosition);
+            }
+        }
```

Comment:
applyMovementStep을 적용하기 위해서 step을 꼭 넘겨줘야할까요? 원하는건 최종 위치가 알맞은지 확인하는거 아닐까요?
```suggestion
    Position movedPosition = movement.apply(position)
    if (!movedPosition.isInvalidPosition() && !board.isExists(movedPosition)) {
        moveBeforeLastStepPositions.put(movement, movePosition);
    }   
```
위와 같이 작성하고 Movement.apply에서 모든 스텝을 적용한 결과물을 반환하면 어떨까요?